storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6969,10 +7103,20 @@ int ha_rocksdb::get_row_by_rowid(uchar *const buf, const char *const rowid,
 
   table->status = STATUS_NOT_FOUND;
   if (found) {
+    /* If we found the record, but it's expired, pretend we didn't find it.  */
+    if (!skip_ttl_check && m_pk_descr->has_ttl() &&
+        should_hide_ttl_rec(rocksdb::Slice(&m_retrieved_record.front(),
+                                           m_retrieved_record.size()),
+                            tx->m_snapshot_timestamp)) {
+      DBUG_RETURN(HA_ERR_KEY_NOT_FOUND);
```

Comment:
as discussed offline, decided to not release the lock to stay consistent with current behaviour.